File: subprojects/frida-node/examples/detached.ts
import frida from "frida";
import readline from "readline";

let input: readline.Interface | null = null;

async function main() {
    const device = await frida.getUsbDevice();
    const session = await device.attach("Hello");
    session.detached.connect(onDetached);

    console.log("[*] Attached. Hit ENTER to exit.");

    input = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
        terminal: true
    });
    input.on("line", () => {
        session.detach();
        input!.close();
    });
}

function onDetached(reason: frida.SessionDetachReason, crash: frida.Crash | null) {
    console.log("[*] onDetached() reason:", reason, "crash:", crash);
    input!.close();
}

main()
    .catch(e => {
        console.error(e);
    });

File: subprojects/frida-python/examples/detached.py
import sys

import frida


def on_detached():
    print("on_detached")


def on_detached_with_reason(reason):
    print("on_detached_with_reason:", reason)


def on_detached_with_varargs(*args):
    print("on_detached_with_varargs:", args)


session = frida.attach("Twitter")
print("attached")
session.on("detached", on_detached)
session.on("detached", on_detached_with_reason)
session.on("detached", on_detached_with_varargs)
sys.stdin.read()
